### PR TITLE
Use a non-greedy regex to match links

### LINKED ISSUE #15: 2 links within a line
Hello,

When I'm creating two links within a sentence (i.e. line), such as  "something something [[link]] something something [[link2]]", the link first delimiter will not be recognized and clicking on the first link will create a new note called
" [[link]] something something [[link2" instead of redirecting to [[link]].

### PR DESCRIPTION
**Pull request by theshoals**
This allows multiple links to work on the same line.

Closes #15

### CODE REVIEW

**Comment by theshoals** [comment]
Note: This fix will cause issues for users who use links like "[[title [specification]]]" (as mentioned in https://github.com/dansheffler/zettelkasten-wiki/issues/4).

I'm not sure if that's an acceptable behavior change.

**Comment by DivineDominion** [comment]
@theshoals What do you think about an optional trailing bracket? See the regex here https://regexr.com/5bat1

    \[\[(.+?[\]]?)\]\]

**Comment by theshoals** [comment]
That would help in the default case.

One complication is that the link-begin and link-end symbols are customizable, so the regex has to be constructed dynamically ("[[" and "]]" are the default symbols).

I wonder if a parser would have to be introduced to handle all cases properly?

**Comment by DivineDominion** [comment]
Hmm if you put the part `(.+?[\]]?)` cannot in the middle instead of `(.+?)` it should  do the trick no matter the actual link beginning/end strings, I think

**Comment by theshoals** [comment]
I was thinking of cases where a character from the customized link-end symbol is present in the link:
- link-begin is "(("
- link-end is "))"
- link-text is "hello (world)"

That would require a different regex from `\[\[(.+?[\]]?)\]\]` to handle the customized link-end symbol.

**Comment by DivineDominion** [comment]
I'd say we worry about that when people begin complaining :) It's rather unlikely that people ditch the double square bracket wiki link convention. If you think regular parens are even worth supporting, `(.+?[\]\)]?)` should do the trick. (This approach of adding optional closing brackets obviously doesn't scale beyond `(.+?[\]\)\}\>]?)` but my point is that whoever thinks that other "brackets" like`lolWikiLinklol` are a good idea should do this themselves)

**Comment by theshoals** [comment]
Thanks for your suggestions.

I think your approach makes sense: If a user customizes the symbols, they can also customize the regex.

The PR has now been updated.

File: lib/wikilink.coffee [hunk 1]
```diff
@@ -26,7 +26,7 @@ module.exports = atomWikiLink =
       title: 'Link Regex'
       description: 'The regex used to identify the contents of a link'
       type: 'string'
-      default: '(.+)'
+      default: '(.+?[\\]]?)'
```
Comment (theshoals) [inline lib/wikilink.coffee:29]: I had to escape the slash (`\\`) for the regex to work.

**Comment by DivineDominion** [comment]
That's great, thanks!

**Pull request merged**